dropwizard-testing/src/test/java/io/dropwizard/testing/DropwizardTestSupportWithResourceConfigProviderTest.java

```diff
@@ -41,4 +47,36 @@ void returnsEnvironment() {
         final Environment environment = TEST_SUPPORT.getEnvironment();
         assertThat(environment.getName()).isEqualTo("TestApplication");
     }
+
+    @Test
+    void doesNotOverwriteConfigSourceProviderIfNotProvidedInConstructor() throws Exception {
+
+        DropwizardTestSupport<TestConfiguration> support = new DropwizardTestSupport<>(
+                TestApplication.class, "test-config.yaml");
+        try {
+            support.before();
+            assertThat(TEST_CONFIG_SOURCE_PROVIDER.openCalled).isTrue();
+        } catch (Exception e) {
+            Assert.fail();
```

Comment:
This catch statement isn't needed, no?

Reply:
Yep, definitely not needed. Pushed a fix.